manifests/params.pp

```diff
@@ -72,46 +72,11 @@
     }
     'ubuntu', 'neon': {
       # Ubuntu: https://ubuntu.com/about/release-cycle and https://wiki.ubuntu.com/Releases
-      case $xfacts['lsbdistcodename'] {
-        # Ubuntu 12.04 LTS in ESM
-        'precise': {
-          $legacy_origin      = true
-          $origins            = [
-            '${distro_id}:${distro_codename}-security', #lint:ignore:single_quote_string_with_variables
-          ]
-
-        }
-        # Ubuntu 14.04 LTS in ESM
-        'trusty': {
-          $legacy_origin      = true
-          $origins            = [
-            '${distro_id}:${distro_codename}-security', #lint:ignore:single_quote_string_with_variables
-          ]
-        }
-        # Ubuntu 16.04 LTS, 18.04 LTS, 20.04 LTS in Standard Support
-        'xenial', 'bionic', 'focal': {
-          $legacy_origin      = true
-          $origins            = [
-            '${distro_id}:${distro_codename}', #lint:ignore:single_quote_string_with_variables
-            '${distro_id}:${distro_codename}-security', #lint:ignore:single_quote_string_with_variables
-          ]
-        }
-        # Ubuntu Interim Releases
-        'eoan': {
-          $legacy_origin      = true
-          $origins            = [
-            '${distro_id}:${distro_codename}', #lint:ignore:single_quote_string_with_variables
-            '${distro_id}:${distro_codename}-security', #lint:ignore:single_quote_string_with_variables
-          ]
-        }
-        default: {
-          warning("Ubuntu ${xfacts['lsbdistrelease']} \"${xfacts['lsbdistcodename']}\" has reached End of Life - please upgrade!")
-          $legacy_origin      = true
-          $origins            = [
-            '${distro_id}:${distro_codename}-security', #lint:ignore:single_quote_string_with_variables
-          ]
-        }
-      }
+      $legacy_origin      = true
+      $origins            = [
+        '${distro_id}:${distro_codename}', #lint:ignore:single_quote_string_with_variables
+        '${distro_id}:${distro_codename}-security', #lint:ignore:single_quote_string_with_variables
+      ]
```

Comment:
All supported versions (20.10, 20.04, 19.10, 18.04, and 16.04) by default also include the origins `${distro_id}ESMApps:${distro_codename}-apps-security` and `${distro_id}ESM:${distro_codename}-infra-security`, maybe they should be added as well? The comment attached to them states;

> // Extended Security Maintenance; doesn't necessarily exist for
> // every release and this system may not have it installed, but if
> // available, the policy for updates is such that unattended-upgrades
> // should also install from here by default.

Reply:
It also lists support for neon but I have no idea about that. Since I personally use neither and this PR only drops old distros, I'd prefer to merge this. If people care about the ESM part, a PR is welcome.

Reply:
Neon is still based very heavily on Ubuntu - except for the KDE-relevant packages, they use the same unattended-upgrade package even in the latest releases.